src/client/sandbox/event/simulator.js

```diff
@@ -105,43 +114,37 @@ export default class EventSimulator {
         return el.dispatchEvent(ev);
     }
 
-    static _getUIEventArgs (type, options) {
-        const opts = options || {};
-
+    static _getUIEventArgs (type, options = {}) {
         return {
             type:       type,
-            canBubble:  opts.canBubble !== false,
-            cancelable: opts.cancelable !== false,
-            view:       opts.view || window,
-            detail:     opts.detail || 0,
-            ctrlKey:    opts.ctrlKey || false,
-            altKey:     opts.altKey || false,
-            shiftKey:   opts.shiftKey || false,
-            metaKey:    opts.metaKey || false
+            canBubble:  options.canBubble !== false,
+            cancelable: options.cancelable !== false,
+            view:       options.view || window,
+            detail:     MOUSE_EVENT_DETAIL[type] || 0,
```

Comment:
We shouldn't lose the `detail` option which is passed to an emulation function.
```js
'detail' in options ? options.detail : MOUSE_EVENT_DETAIL[type]
```

Reply:
How about this one?
```javascript
static _getUIEventArgs (type, options = {}) {
    const detail = 'detail' in options ? options.detail : MOUSE_EVENT_DETAIL[type];

    return {
        type:       type,
        canBubble:  options.canBubble !== false,
        cancelable: options.cancelable !== false,
        view:       options.view || window,
        detail:     detail || 0,
        ctrlKey:    options.ctrlKey || false,
        altKey:     options.altKey || false,
        shiftKey:   options.shiftKey || false,
        metaKey:    options.metaKey || false
    };
}
```